### PR TITLE
Refactor PaginatedContainer to avoid code duplication

### PR DESCRIPTION
**Pull request by EVAST9919**
Split from https://github.com/ppy/osu/pull/5778

### CODE REVIEW

File: osu.Game/Overlays/Profile/Sections/PaginatedContainer.cs [hunk 1]
```diff
@@ -11,22 +11,27 @@
...
 using osu.Game.Rulesets;
 using osu.Game.Users;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
 
 namespace osu.Game.Overlays.Profile.Sections
 {
-    public abstract class PaginatedContainer : FillFlowContainer
+    public abstract class PaginatedContainer<T> : FillFlowContainer
```
Comment (iiSaLMaN) [inline osu.Game/Overlays/Profile/Sections/PaginatedContainer.cs]: Ensure given argument is an `APIRequest`
```suggestion
    public abstract class PaginatedContainer<T> : FillFlowContainer
        where T : APIRequest
```

Comment (iiSaLMaN) [inline osu.Game/Overlays/Profile/Sections/PaginatedContainer.cs]:
```suggestion
    public abstract class PaginatedContainer<TRequest> : FillFlowContainer
```

Comment (EVAST9919) [reply osu.Game/Overlays/Profile/Sections/PaginatedContainer.cs]: Please read the code. T is not an api request. T is response of created request.

Comment (iiSaLMaN) [reply osu.Game/Overlays/Profile/Sections/PaginatedContainer.cs]: oh sorry, nvm this then

Comment (EVAST9919) [reply osu.Game/Overlays/Profile/Sections/PaginatedContainer.cs]: May be `TAPIResponse` would be better

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**